source/Genome.cpp

```diff
@@ -43,7 +43,7 @@ Genome::~Genome()
 void Genome::freeMemory(){//free big chunks of memory used by genome and suffix array
 
     if (pGe.gLoad=="NoSharedMemory") {//can deallocate only for non-shared memory
-        delete[] G1;
+        if (G1 != NULL) delete[] G1;
```

Comment:
It is safe to delete a `null` pointer and common style is to remove the redundant `if`.
See https://stackoverflow.com/questions/4190703/is-it-safe-to-delete-a-null-pointer